### PR TITLE
fix: comment box input focus on the end

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- This fixes the issue when replying to someone and mention is included, focus will be at the end of the content.
- This also fixes the issue when you edit a comment, the focus would automatically be then at the end.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1896 #done

### CODE REVIEW

**Review by rebelchris — APPROVED** [review]

**Review by capJavert — COMMENTED** [review]
I see more and more hacks with inputs like this due to not having reactive state instance in recent times. I get it we don't have other option now but it looks like a lot of tech debt.

File: packages/shared/src/components/fields/MarkdownInput/index.tsx [hunk 1]
```diff
@@ -150,6 +152,16 @@ function MarkdownInput(
...
 
+  useEffect(() => {
+    const content = textareaRef?.current?.value;
+
+    if (!content) {
+      return;
+    }
+
+    focusInput(textareaRef.current, [content.length, content.length]);
+  }, []);
```
Comment (capJavert) [inline packages/shared/src/components/fields/MarkdownInput/index.tsx:163]: I see more and more hacks with inputs like this due to not having reactive state instance in recent times. I get it we don't have other option now but it looks like a lot of tech debt.

Comment (rebelchris) [reply packages/shared/src/components/fields/MarkdownInput/index.tsx:163]: Very valid point, for now lets proceed, ill add it as discussion topic for web. Around moving to form lib. (See ido also mentioned one)

Is that ok?

Comment (capJavert) [reply packages/shared/src/components/fields/MarkdownInput/index.tsx:163]: Yes of course, don't think of it as blocking.

**Review by omBratteng — APPROVED** [review]

File: packages/shared/src/components/fields/MarkdownInput/index.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/fields/MarkdownInput/index.tsx:163]: Most likely a form library.
There are some suggestions floating for instance https://tanstack.com/form/latest

Comment (sshanzel) [reply packages/shared/src/components/fields/MarkdownInput/index.tsx:163]: I see. Yep. I would be glad if we could use any available packages like that, or `react-form-hooks` can be a good option as well. We can totally refactor our input components by then.

**Pull request merged**